### PR TITLE
Adds IgnoreOnError as an errorHandling strategy

### PR DESCRIPTION
**Pull request by fabianofranz**
The `IgnoreOnError` strategy allows the flag parser to continue parsing in case of error (e.g. flag provided but not declared). In that strategy, flags that fail to parse will just be considered arguments which can be retrieved by `Args()` in the flag set.

ptal @eparis @spf13

### CODE REVIEW

**Comment by eparis** [comment]
Is there never a case one would want IgnoreOnError but would still want to know $somehow that an error occurred?  I guess we could merge this, and if such a case ever sprung up we could store the error(s) somewhere in a new field in the FlagSet. Maybe?

I hate APIs that lose data, but I'm not sure of a good way not to lose data while doing what you need to do...

File: flag.go [hunk 1]
```diff
@@ -1005,7 +1007,11 @@ func (f *FlagSet) parseArgs(args []string, fn parseFunc) (err error) {
 			args, err = f.parseShortArg(s, args, fn)
 		}
 		if err != nil {
-			return
+			if f.errorHandling != IgnoreOnError {
```
Comment (eparis) [inline flag.go]: I'm also interested if there is a way to push this error handling into `Parse()` so all of them are handled together (making code maintenance easier/more obvious) but I'm not seeing a great way to do that either...

**Comment by fabianofranz** [comment]
@eparis We don't have such case as of now, but I'm ok with adding that in future if there's any need. Would probably need an additional structure to store it like you mentioned. 

FWIW, the current mindset is that this strategy new consider an arg anything that failed to parse as a flag, and like any other arg you don't store the parsing failure reason.

File: flag.go (hunk 1 above)
Comment (fabianofranz) [inline flag.go]: Yeah, I wanted that in the first place too... but it has to be checked on every individual arg, and `Parse()` only knows the accumulated result.

**Comment by eparis** [comment]
would you be willing to apply https://github.com/eparis/pflag/commit/e70fde5b329413e98a48d6f98f8f74cb2f0bc339.patch and work through and/or document the logic? Maybe add some more shorthand tests?

**Comment by fabianofranz** [comment]
I'm closing this for now to come up with a better strategy. Not feeling confident about the amount of corner cases and unpredictability here.
